File: LeetCode/[M] - 1143. Longest Common Subsequence/solution.py
class Solution:
    def longestCommonSubsequence(self, text1: str, text2: str) -> int:
        prev, curr = [0] * (len(text1) + 1), [0] * (len(text1) + 1)

        for i in range(1, len(text2) + 1):
            for j in range(1, len(text1) + 1):
                if text2[i - 1] == text1[j - 1]:
                    curr[j] = prev[j - 1] + 1
                else:
                    curr[j] = max(curr[j - 1], prev[j])
            prev, curr = curr, [0] * (len(text1) + 1)

        return prev[-1]

File: LeetCode/[M] - 1143. Longest Common Subsequence/solution.go
package main

func longestCommonSubsequence(text1 string, text2 string) int {
	prev, curr := make([]int, len(text1)+1), make([]int, len(text1)+1)
	for i := 1; i < len(text2)+1; i++ {
		for j := 1; j < len(text1)+1; j++ {
			if text1[j-1] == text2[i-1] {
				curr[j] = prev[j-1] + 1
			} else {
				curr[j] = max(curr[j-1], prev[j])
			}
		}
		prev, curr = curr, make([]int, len(text1)+1)
	}
	return prev[len(prev)-1]
}
